### PR TITLE
debian: exclude various unneeded build artifacts from scst-dkms

### PR DESCRIPTION
**Pull request by scop**
scst-dkms is currently unnecessarily large, as it contains a dirty, built source tree with various build artifacts that are not needed in the package. This excludes a bunch of them.

### CODE REVIEW

**Comment by bvanassche** [comment]
How about moving these changes from debian/rules into scripts/list-source-files (see also `( cd "$d" && find . -type f -o -type l | sed 's|^\./||' )` in that file)? I think that the files that you want to exclude only can get included if the SCST source code is not under source control. As one can see in scripts/list-source-files, only files that are managed by the source control system are included if Subversion or git metadata is found.

**Comment by scop** [comment]
Sounds good to me, will have a look. In the meantime, revised the initial patch -- it was removing `conftest` entirely which obviously isn't desirable.

**Comment by scop** [comment]
Ok, done. I added some more exclusions but didn't feel like duplicating the entire `.gitignore` or the like.

A better option could be to include the list of files already generated and processed by `make-scst-dist` in the resulting tarballs as-is, and make scripts/list-source-files use it if present and no scm dir is.

File: scripts/list-source-files [hunk 1]
```diff
@@ -39,7 +39,17 @@ list_source_files() {
       hg manifest
     fi
   else
-    ( cd "$d" && find . -type f -o -type l | sed 's|^\./||' )
+    (
+      cd "$d" &&
+      find . -type f -o -type l | \
+      sed -e 's|^\./||' \
+	  -e '\|\.k\{0,1\}o\(\.\(cm\)\{0,1\}d\)\{0,1\}$|d' \
```
Comment (bvanassche) [inline scripts/list-source-files:46]: Are the backslashes past the opening quote ('\...) needed? If not, please leave these out.

Comment (scop) [reply scripts/list-source-files:46]: They are, for everything besides a `/` (which would work too, but requires more backslashing here).

https://pubs.opengroup.org/onlinepubs/9699919799/utilities/sed.html#tag_20_116_13_02

```shell
$ printf "%s\n" foo bar | sed -e '|foo|d'
sed: -e expression #1, char 1: unknown command: `|'
$ printf "%s\n" foo bar | sed -e '\|foo|d'
bar
```

Comment (scop) [reply scripts/list-source-files:46]: By the way, the regexps could be simplified quite a bit if you're willing to assume a sed implementation that has the `-r` option. It's not POSIX, but GNU sed, busybox sed, and ssed all do have it (even FreeBSD sed does too), I'm not sure if there are any relevant sed implementations out in existence that don't.

**Comment by bvanassche** [comment]
Thank you for the additional clarification, and thanks for the patch!

**Pull request merged**